src/app/page.tsx

```diff
@@ -1,91 +1,97 @@
 "use client";
 
 import { useEffect, useState } from "react";
+import SearchBar from "@/app/components/SearchBar";
+import SearchStatus from "@/app/components/SearchStatus";
+import AdvocatesTable from "@/app/components/AdvocatesTable";
+import { Advocate } from "@/app/types";
 
 export default function Home() {
-  const [advocates, setAdvocates] = useState([]);
-  const [filteredAdvocates, setFilteredAdvocates] = useState([]);
+  const [advocates, setAdvocates] = useState<Advocate[]>([]);
+  const [filteredAdvocates, setFilteredAdvocates] = useState<Advocate[]>([]);
+  const [searchTerm, setSearchTerm] = useState("");
```

Comment:
Fixes: #2 Direct DOM manipulation in `page.tsx`

The direct DOM manipulation of `#search-term` _document.getElementById("search-term").innerHTML = searchTerm_ is anti-React and bypasses React's virtual DOM ultimately breaking its rendering flow. By introducing `useState`, React automatically re-renders and updates the DOM where it's used.

`searchTerm` is updated whenever `setSearchTerm` is called. In this case, it's called in 2 places - 
1) whenever the search input value changes
2) whenever the `Reset Search` button is pressed.